internal/internal.go

```diff
@@ -223,24 +219,24 @@ func (i *Internal) VICELaunchApp(writer http.ResponseWriter, request *http.Reque
 
 	buf, err := ioutil.ReadAll(request.Body)
 	if err != nil {
-		http.Error(writer, err.Error(), http.StatusBadRequest)
+		common.Error(writer, err.Error(), http.StatusBadRequest)
 		return
 	}
 
 	if err = json.Unmarshal(buf, job); err != nil {
-		http.Error(writer, err.Error(), http.StatusBadRequest)
+		common.Error(writer, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	if err = i.validateJob(job); err != nil {
-		http.Error(writer, err.Error(), http.StatusBadRequest)
+	if status, err := i.validateJob(job); err != nil {
+		common.DetailedError(writer, err, status)
```

Comment:
`validateJob` returns both errors that indicate a bad request and errors that indicate that something is broken. There wasn't a good way to distinguish them before, so I added the HTTP status as a return value. There's a huge part of me that hates that because it breaks the transparency that I mentioned before. On the other hand, putting the HTTP status in the error itself doesn't really seem right either.

I'm still on the fence about this one. If someone has a strong preference or other ideas, please feel free to let me know.